variable interpreted as type

### The following program `sample.go` triggers an unexpected result

```go
package main

import "fmt"

func check() (result bool, err error) {                                                                                                           
    return true, nil                                                                                                                              
}                                                                                                                                                 
                                                                                                                                                  
func main() {                                                                                                                                     
    result, error := check()                                                                                                                        
    fmt.Println(result, error)                                                                                                                      
}
```


### Expected result

```console
$ go run ./sample.go
true <nil>
```


### Got

```console
$ yaegi ./sample.go
./sample:8:2: assignment mismatch: 1 variables but check returns 2 values
```


### Yaegi Version

v0.10.0

### Additional Notes

This issue was found when testing gopkg.in/check:

```
$ yaegi test gopkg.in/check.v1
test: /Users/marc/go/src/gopkg.in/check.v1/benchmark_test.go:7:2: import "gopkg.in/check.v1" error: /Users/marc/go/src/gopkg.in/check.v1/helpers.go:216:2: assignment mismatch: 1 variables but  returns 2 values
```

--- a/_test/issue-1306.go
+++ b/_test/issue-1306.go
@@ -1,0 +1,15 @@
+package main
+
+import "fmt"
+
+func check() (result bool, err error) {
+	return true, nil
+}
+
+func main() {
+	result, error := check()
+	fmt.Println(result, error)
+}
+
+// Output:
+// true <nil>

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -2013,7 +2013,7 @@ func compDefineX(sc *scope, n *node) error {
 		} else {
 			types = funtype.ret
 		}
-		if n.child[l-1].isType(sc) {
+		if n.anc.kind == varDecl && n.child[l-1].isType(sc) {
 			l--
 		}
 		if len(types) != l {

